atomic_reactor/plugins/build_buildah.py

```diff
@@ -0,0 +1,80 @@
+"""
+Copyright (c) 2019 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+from __future__ import print_function, unicode_literals
+
+import subprocess
+from six import PY2
+import os
+
+from atomic_reactor.util import get_exported_image_metadata
+from atomic_reactor.plugin import BuildStepPlugin
+from atomic_reactor.build import BuildResult
+from atomic_reactor.constants import CONTAINER_BUILDAH_BUILD_METHOD
+from atomic_reactor.constants import EXPORTED_SQUASHED_IMAGE_NAME, IMAGE_TYPE_DOCKER_ARCHIVE
+
+
+class BuildahPlugin(BuildStepPlugin):
+    """
+    Build image using buildah https://buildah.io/
+    This requires the buildah executable binary to be in $PATH.
+    """
+
+    key = CONTAINER_BUILDAH_BUILD_METHOD
+
+    def run(self):
+        """
+        Build image inside current environment using buildah;
+        It's expected this may run within (privileged) oci container.
+
+        Returns:
+            BuildResult
+        """
```

Comment:
There is quite a lot more work needed to make this functional, because 'buildah bud' creates an image available to podman but not docker.

We're planning to do the podman integration work soon, but for now can we make this build-step plugin raise `NotImplementedError` or something in `run()`? That way it won't get used accidentally.